### ISSUE TITLE
Python runtime crashes on invalid arguments

### ISSUE THREAD

**Issue by pcyin** [clab/dynet#938]
There are still cases where the python runtime crashes with invalid arguments. One example is:
```
import dynet as dy

model = dy.ParameterCollection()
rnn = dy.LSTMBuilder(1, 2, 3, model)
s0 = rnn.initial_state()
s0.set_h()
```

**Comment by pmichel31415** [comment]
Should be fixed by #939 , let me know if you find more occurrences of the same problem

**Comment by neubig** [comment]
Should be fixed by https://github.com/clab/dynet/pull/939
